starlette/middleware/__init__.py

```diff
@@ -38,5 +36,6 @@ def __repr__(self) -> str:
         class_name = self.__class__.__name__
         args_strings = [f"{value!r}" for value in self.args]
         option_strings = [f"{key}={value!r}" for key, value in self.kwargs.items()]
-        args_repr = ", ".join([self.cls.__name__] + args_strings + option_strings)
+        cls_name = self.cls.__name__  # type: ignore[attr-defined]
```

Comment:
Is this safe to do? I guess the point is that the `self.cls` isn't necessarily a class, it may be a function, but even functions have `__name__` defined. Maybe we should do `getattr(self.cls, "__name__", "")` or something to make sure we don't error in any case?

Reply:
(alternatively we could add `__name__` to the protocol above 😆 )

Reply:
@adriangb Thanks for suggestion, changed it to be `getattr` with default value call.

Reply:
Thanks! Are there any actual changes to the reprs? I'm not seeing why that code had to change (aside from what I pointed out in this comment)

Reply:
There were no actual changes to the repr. I changed it only because of mypy error.